Unix/provmgr/provmgr.c

```diff
@@ -431,7 +431,7 @@ static MI_Result MI_CALL _GetProviderByClassName(
         if (!lib)
         {
             trace_OpenProviderLib_Failed(scs(proventry->libraryName));
-            return MI_RESULT_FAILED;
+            return MI_RESULT_FAIL_LOAD_PROVIDER;
```

Comment:
So here is where the error is reported. The problem is the only way today to report an error from this code is an error code. Instead we need a way of reporting back an actual error... where an error is more than just an error code, but also an error string and an error category. Maybe that is an out parameter to the function that returns an OMI_Error instead and if one is returned then that error object is used rather than just the error.